### PR TITLE
[Feat] Support scalar types in AdditionalInformationEntry

### PR DESCRIPTION
**Pull request by NickCao**
## Purpose

Support scalar types in AdditionalInformationEntry, also factor out the code for deserializing these entries.

## Test Plan

```bash
pytest vllm_omni/engine/test_serialization.py -v
python examples/offline_inference/qwen3_tts/end2end.py --query-type CustomVoice --output-dir /tmp/test
```

## Test Result

```
vllm_omni/engine/test_serialization.py::test_serialize_additional_information PASSED [100%]
Processed prompts: 100%|██████████| 1/1 [00:24<00:00, 24.91s/it]
# and the output audio sounds correct
```
---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan. Please provide the test scripts & test commands. Please state the reasons if your codes don't require additional test scripts. For test file guidelines, please check the [test style doc](https://docs.vllm.ai/projects/vllm-omni/en/latest/contributing/ci/tests_style/)
- [x] The test results. Please paste the results comparison before and after, or the e2e results.
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model. **Please run `mkdocs serve` to sync the documentation editions to `./docs`.**
- [ ] (Optional) Release notes update. If your change is user-facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm_omni/engine/serialization.py [hunk 1]
```diff
@@ -64,18 +65,49 @@ def serialize_additional_information(
...
-                type(value).__name__,
-            )
-        else:
-            logger.warning(
-                "[%s] Dropping unsupported additional_information key=%s type=%s",
-                log_prefix,
-                key,
-                type(value).__name__,
-            )
+        entries[key] = AdditionalInformationEntry(scalar_data=value)
```
[the remark this answers is not in the document]
Comment (NickCao) [reply vllm_omni/engine/serialization.py:68]: Previously list values can also contain types that can't be serialized by msgspec, this remains the same.

**Comment by hsliuustc0106** [comment]
## Review Summary

**Verdict**: ✅ LGTM (No Blockers)

Clean refactoring with good test coverage. Nice reduction of duplicated deserialization code!

### Blocker Scan

| Category | Result |
|----------|--------|
| Correctness | ✅ PASS - Proper handling of tensor/list/scalar forms |
| Reliability/Safety | ✅ PASS - Exception handling with logging |
| Breaking Changes | ✅ PASS - Backward compatible (adds field) |
| Test Coverage | ✅ PASS - Unit test + E2E test provided |
| Documentation | ✅ PASS - Good docstrings on new function |
| Security | ✅ PASS - No security issues |

---

### What Was Validated

✅ **Feature correctness**: Added `scalar_data` field to `AdditionalInformationEntry`
✅ **Code deduplication**: Extracted `deserialize_additional_information()` into shared function
✅ **Test coverage**: Unit test validates tensor/list/scalar serialization round-trip
✅ **E2E validation**: Qwen3 TTS end-to-end test passes

---

### Code Quality Highlights

**serialization.py:73-115** - Excellent refactoring! This function replaces duplicate code in:
- `omni_ar_scheduler.py:528`
- `gpu_model_runner.py:933`

The new function is more complete:
- ✅ Handles all 3 forms (tensor/list/scalar)
- ✅ Duck-typed to accept any object with `.entries`
- ✅ Consistent error handling

**\_\_init\_\_.py:45-47** - Clean schema extension. Adding `scalar_data` is backward compatible since msgspec ignores `None` fields.

**serialization.py:70-72** - Previously logged warnings for unsupported types. Now preserves them as scalars. This is an improvement (no data loss).

---

### Minor Observation (Non-blocking)

**test_serialization.py:32-34** - The test deletes tensors before comparing dicts:

```python
del deserialized["tensor_f32"], info["tensor_f32"]
del deserialized["tensor_f64"], info["tensor_f64"]
assert deserialized == info
```

This works, but could be clearer with explicit assertions:

```python
assert deserialized["list"] == info["list"]
assert deserialized["scalar"] == info["scalar"]
```

(Not requesting changes — the current test is correct.)

---

Ready to merge.

**Comment by linyueqian** [comment]
Thanks for the cleanup — consolidating the 3 copies of deserialization logic into `deserialize_additional_information()` is a nice improvement, and the scalar support makes sense for passing values like `ref_code_len` between stages (as noted in your [comment on #2104](https://github.com/vllm-project/vllm-omni/pull/2104#issuecomment-4113224263)).

LGTM.

**Comment by NickCao** [comment]
@linyueqian anything blocking the auto merge?

File: vllm_omni/engine/test_serialization.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import torch
```
Comment (hsliuustc0106) [inline vllm_omni/engine/test_serialization.py:1]: RM THE TEST UNDER ENGINE CC @linyueqian

Comment (linyueqian) [reply vllm_omni/engine/test_serialization.py:1]: ok
